src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -323,8 +328,9 @@ def __new_assessment_summary_json(self, assessment_packages_json, status, code):
             else:
                 other_patch_count += 1
 
-        # discern started by
-        started_by = Constants.PatchAssessmentSummaryStartedBy.PLATFORM if self.execution_config.exec_auto_assess_only else Constants.PatchAssessmentSummaryStartedBy.USER
+        # discern started by - either pure auto-assessment or assessment data being included with configure patching with assessmentMode set to AutomaticByPlatform
+        include_assessment_with_configure_patching = (self.execution_config.operation == Constants.CONFIGURE_PATCHING and self.execution_config.assessment_mode == Constants.AssessmentModes.AUTOMATIC_BY_PLATFORM)
+        started_by = Constants.PatchAssessmentSummaryStartedBy.PLATFORM if (self.execution_config.exec_auto_assess_only or include_assessment_with_configure_patching) else Constants.PatchAssessmentSummaryStartedBy.USER
 
```

Comment:
With these checks changing overtime, can we have a centralized place which defines what happens when? For eg: in this case, include_assessment_with_configure_patching can be computed in some common location and then be re-used in multiple places (status updates, started_by, etc)

Reply:
Will refactor this in the context of the bigger PR (that started later): https://github.com/Azure/LinuxPatchExtension/pull/183 
Plan is to take it into a 'derived' field in execution config and fan out from there.